src/main/java/org/openrewrite/java/security/xml/TransformerFactoryInsertAttributeStatementVisitor.java

```diff
@@ -15,76 +15,45 @@
-            attributeTemplate.append(transformerFactoryVariableName).append(".setAttribute(XMLConstants.ACCESS_EXTERNAL_STYLESHEET, \"\");");
+            getTemplate().append(getFactoryVariableName()).append(".setAttribute(XMLConstants.ACCESS_EXTERNAL_STYLESHEET, \"\");");
         }
         if (needsFeatureSecureProcessing) {
-            attributeTemplate.append(transformerFactoryVariableName).append(".setFeature(XMLConstants.FEATURE_SECURE_PROCESSING, true);");
+            getTemplate().append(getFactoryVariableName()).append(".setFeature(XMLConstants.FEATURE_SECURE_PROCESSING, true);");
         }
     }
 
     @Override
     public J.Block visitBlock(J.Block block, P ctx) {
         J.Block b = super.visitBlock(block, ctx);
-        Statement beforeStatement = null;
-        if (b.isScope(scope)) {
-            for (int i = b.getStatements().size() - 2; i > -1; i--) {
-                Statement st = b.getStatements().get(i);
-                Statement stBefore = b.getStatements().get(i + 1);
-                if (st instanceof J.MethodInvocation) {
-                    J.MethodInvocation m = (J.MethodInvocation) st;
-                    if (TransformerFactoryFixVisitor.TRANSFORMER_FACTORY_INSTANCE.matches(m) || TransformerFactoryFixVisitor.TRANSFORMER_FACTORY_SET_ATTRIBUTE.matches(m)) {
-                        beforeStatement = stBefore;
-                    }
-                } else if (st instanceof J.VariableDeclarations) {
-                    J.VariableDeclarations vd = (J.VariableDeclarations) st;
-                    if (vd.getVariables().get(0).getInitializer() instanceof J.MethodInvocation) {
-                        J.MethodInvocation m = (J.MethodInvocation) vd.getVariables().get(0).getInitializer();
-                        if (m != null && TransformerFactoryFixVisitor.TRANSFORMER_FACTORY_INSTANCE.matches(m)) {
-                            beforeStatement = stBefore;
-                        }
-                    }
-                }
-            }
-
-            if (getCursor().getParent() != null && getCursor().getParent().getValue() instanceof J.ClassDeclaration) {
-                attributeTemplate.insert(0, "{").append("}");
-            }
-            JavaCoordinates insertCoordinates = beforeStatement != null ?
-                    beforeStatement.getCoordinates().before() :
-                    b.getCoordinates().lastStatement();
-            b = JavaTemplate
-                    .builder(attributeTemplate.toString())
-                    .imports("javax.xml.XMLConstants")
-                    .contextSensitive()
-                    .build()
-                    .apply(new Cursor(getCursor().getParent(), b), insertCoordinates);
-            maybeAddImport("javax.xml.XMLConstants");
+        Statement beforeStatement = getInsertStatement(b);
+        if (b.isScope(getScope())) {
+            b = updateTemplate(b, block, beforeStatement, Stream.of("javax.xml.XMLConstants").collect(Collectors.toSet()));
```

Comment:
```suggestion
            b = updateTemplate(b, block, beforeStatement, Collections.single("javax.xml.XMLConstants"));
```

Not sure I got the method name entirely correct